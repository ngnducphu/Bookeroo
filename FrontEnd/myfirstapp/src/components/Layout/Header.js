import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { searchBooks } from "../../actions/bookActions";
import { syncLocalStorage } from "../../actions/cartActions";
import { useHistory, useLocation } from "react-router";
import { Badge } from "reactstrap";
const Header = (props) => {
  const location = useLocation();

  useEffect(() => {
    props.syncLocalStorage();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  const history = useHistory();
  const handleOptionSelect = (e) => {
    e.preventDefault();
    const searchTerm = e.currentTarget.text;
    props.searchBooks("category", searchTerm, history);
  };

  return (
    <div>
      <nav className="navbar navbar-expand-sm navbar-dark bg-primary mb-4">
        <div className="container">
          <Link className="navbar-brand" to="/main">
            Bookeroo
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                <Link className="nav-link" to="/aboutus">
                  About us
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="/contactus">
                  Contact us
                </Link>
              </li>
              {location.pathname === "/main" ? (
                <li className="nav-item">
                  <div className="dropdown">
                    <a
                      className="nav-link dropdown-toggle"
                      role="button"
                      id="dropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                      href="https://www.bookdepository.com/"
                      onClick={(e) => {
                        e.currentTarget.click();
                      }}
                    >
                      Category
                    </a>

                    <div
                      className="dropdown-menu"
                      aria-labelledby="dropdownMenuLink"
                    >
                      <a
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                        href="https://www.bookdepository.com/"
                      >
                        Science Fiction
                      </a>
                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Novel
                      </a>

                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Fantasy
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Thriller
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Romance
                      </a>
                    </div>
                  </div>
                </li>
              ) : (
                ""
              )}
            </ul>
            {props.security.validToken !== true ? (
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link className="nav-link " to="/register">
                    Sign Up
                  </Link>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" to="/login">
                    Login
                  </Link>
                </li>
              </ul>
            ) : (
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link className="nav-link " to="/shoppingcart">
                    <i className="fa fa-shopping-cart ">
                      <Badge variant="info">{props.cart.length} </Badge>
                    </i>
                  </Link>
                </li>
                &nbsp;&nbsp;&nbsp;&nbsp;
                <li className="nav-item">
                  <Link className="nav-link " to="/profile">
                    My Profile
                  </Link>
                </li>
                {props.security.user.role === "ROLE_ADMIN" ? (
                  <li className="nav-item">
                    <Link className="nav-link " to="/admin">
                      Admin
                    </Link>
                  </li>
                ) : (
                  ""
                )}
              </ul>
            )}
          </div>
        </div>
      </nav>
    </div>
  );
};

const mapStateToProps = (state) => ({
  security: state.security,
  cart: state.cartState.cart,
});
export default connect(mapStateToProps, { searchBooks, syncLocalStorage })(
  Header
);

export const HeaderPlain = () => {
  return (
    <div>
      <nav className="navbar navbar-expand-sm navbar-dark bg-primary mb-4">
        <div className="container">
          <Link className="navbar-brand" to="/main">
            Bookeroo
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                <Link className="nav-link" to="/aboutus">
                  About us
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="/contactus">
                  Contact us
                </Link>
              </li>
              {"testing" === "/main" ? (
                <li className="nav-item">
                  <div className="dropdown">
                    <a
                      className="nav-link dropdown-toggle"
                      role="button"
                      id="dropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                      href="https://www.bookdepository.com/"
                      onClick={(e) => {
                        e.currentTarget.click();
                      }}
                    >
                      Category
                    </a>

                    <div
                      className="dropdown-menu"
                      aria-labelledby="dropdownMenuLink"
                    >
                      <a
                        className="dropdown-item"
                        href="https://www.bookdepository.com/"
                      >
                        Science Fiction
                      </a>
                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Novel
                      </a>

                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Fantasy
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Thriller
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Romance
                      </a>
                    </div>
                  </div>
                </li>
              ) : (
                ""
              )}
            </ul>

            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <Link className="nav-link " to="/profile">
                  My Profile
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
};
