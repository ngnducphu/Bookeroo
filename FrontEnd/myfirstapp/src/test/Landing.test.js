import React from "react";
import { render, cleanup } from "@testing-library/react";
import { LandingPlain } from "../components/Layout/Landing";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
afterEach(cleanup);

test("Render without crashing", () => {
  render(
    <BrowserRouter>
      <LandingPlain />
    </BrowserRouter>
  );
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <LandingPlain />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
