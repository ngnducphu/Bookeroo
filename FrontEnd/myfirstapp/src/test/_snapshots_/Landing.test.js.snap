// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="landing"
>
  <div
    className="light-overlay landing-inner text-dark"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 text-center"
        >
          <h1
            className="display-3 mb-4"
          >
            Person Management Tool
          </h1>
          <p
            className="lead"
          >
            Create your account to join
          </p>
          <hr />
          <a
            className="btn btn-lg btn-primary mr-2"
            href="/register"
            onClick={[Function]}
          >
            Sign Up
          </a>
          <a
            className="btn btn-lg btn-secondary mr-2"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
